hack/test-go.sh

```diff
@@ -303,6 +309,9 @@ reportCoverageToCoveralls() {
   fi
 }
 
+# Start etcd
+kube::etcd::start
```

Comment:
And only in tests that need this (I think such unit tests are only in pkg/registry and pkg/storage)